core-processor/src/ext.rs

```diff
@@ -1218,12 +1343,15 @@ impl Externalities for Ext {
         message_id: MessageId,
         amount: u64,
     ) -> Result<(), Self::FallibleError> {
-        self.reduce_gas(amount)?;
+        let mut changes = ExtChanges::new(self);
+        changes.reduce_gas(amount)?;
 
         self.context
             .message_context
             .reply_deposit(message_id, amount)?;
 
+        changes.apply(self);
```

Comment:
Im wondering how to guarantee that we didn't forget to call changes.apply?

Reply:
One way is to introduce `applied: bool` field and if it's false on `Drop for ExtChanges` panic that there's unapplied changes. Another way is to somehow have `Changes::with_changes(|changes| { ... })` and it could store both `Ext` we modify and changes that we're about to make

Reply:
Discussed in dm